src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorIT.java

```diff
@@ -279,4 +283,64 @@ public void testBackwardsCompatibilityDataStream() throws Exception {
     container = ElasticsearchContainer.fromSystemProperties();
     container.start();
   }
+
+  @Test
+  public void testRoutingSmtSynchronousMode() throws Exception {
+    props.put("transforms", "TimestampRouter");
+    props.put("transforms.TimestampRouter.type", "org.apache.kafka.connect.transforms.TimestampRouter");
+    String timestampFormat = "YYYYMM";
+    SimpleDateFormat formatter = new SimpleDateFormat(timestampFormat);
+    Date date = new Date(System.currentTimeMillis());
+    props.put("transforms.TimestampRouter.topic.format", "route-it-to-here-${topic}-at-${timestamp}");
+    props.put("transforms.TimestampRouter.timestamp.format", timestampFormat);
+    props.put(FLUSH_SYNCHRONOUSLY_CONFIG, "true");
+    index = String.format("route-it-to-here-%s-at-%s", TOPIC, formatter.format(date));
+    runSimpleTest(props);
+    waitForCommittedOffsets(CONNECTOR_NAME, TOPIC, 0, NUM_RECORDS);
+  }
+
+  @Test
+  public void testRoutingSmtAsynchronousMode() throws Exception {
+    props.put("transforms", "TimestampRouter");
```

Comment:
nit: you can reduce some code duplication with a helper method to set up and run the test, then have the testcase just call the helper method with the configuration to test. Even better would be to parameterize the testcase with junit5 `@ParameterizedTest` feature.

Reply:
Thanks, extracted common code into `addRoutingSmt` method.